docs/contributor/arch/009-makefile-architecture.md

```diff
@@ -0,0 +1,40 @@
+# 9. Makefile: Simplified Architecture
+
+Date: 2024-28-02
+
+## Status
+
+Proposed
+
+## Context: Cluttered Makefile Targets
+
+Makefile targets are currently cluttered, unorganized, and redundant (see the dependency graph bellow). Targets are not parametrized, thus resulting in multiple make targets serving the same scope (e.g. running e2e tests). Targets are also used simultaneously by developers/contributors, as well as within GitHub actions and shell scripts, which makes dependencies even more intertwined. These factors result in a complex environment that is becoming hard to manage by the developers.
+
+![Makefile Dependency Graph (After)](../assets/makefile-graph-before.png)
+
+## Decision
+
+After analyzing repositories that have similar configurations as telemetry-manager (e.g. Kubebuilder default configuration, Kyma modules repositories, other Go repositories), the following decisions have been made (see image bellow):
+
+![Makefile Archtiecture](../assets/makefile-architecture.jpg)
+
+1. Split general Makefile into sub-makefiles:
```

Comment:
excellent use of ordered list (subsequent steps) and unordered list items, even with correct indentation!